src/transformers/trainer.py

```diff
@@ -607,52 +561,18 @@ def __init__(
                         "but SageMaker Model Parallelism < 1.10 does not support FP16 in trainer."
                     )
 
-        if (args.fp16 or args.bf16) and self.sharded_ddp is not None:
-            if args.half_precision_backend == "auto":
-                if args.device == torch.device("cpu"):
-                    if args.fp16:
-                        raise ValueError("Tried to use `fp16` but it is not supported on cpu")
-                    else:
-                        args.half_precision_backend = "cpu_amp"
-                else:
-                    args.half_precision_backend = "cuda_amp"
-
-            logger.info(f"Using {args.half_precision_backend} half precision backend")
-
-        self.do_grad_scaling = False
+        # Apex Mixed precision setup
+        self.use_apex = False
         if (args.fp16 or args.bf16) and not (self.is_deepspeed_enabled or is_sagemaker_mp_enabled()):
             # deepspeed and SageMaker Model Parallel manage their own half precision
-            if self.sharded_ddp is not None:
-                if args.half_precision_backend == "cuda_amp":
-                    self.use_cuda_amp = True
```

Comment:
mix precision logic is all moved to accelerate, so these codes do not take effect any longer when removing SharededDDP.